tests/e2e/specs/wcpay/shopper/shopper-bnpls-checkout.spec.js

```diff
@@ -17,55 +17,82 @@ const UPE_METHOD_CHECKBOXES = [
 	"//label[contains(text(), 'Afterpay')]/preceding-sibling::span/input[@type='checkbox']", // afterpay
 ];
 
-describe( 'BNPL checkout', () => {
-	beforeAll( async () => {
-		await merchant.login();
-		await merchantWCP.enablePaymentMethod( UPE_METHOD_CHECKBOXES );
-		await merchant.logout();
-		await shopper.login();
-		await shopperWCP.changeAccountCurrencyTo( 'USD' );
-	} );
-
-	afterAll( async () => {
-		await shopperWCP.logout();
-		await merchant.login();
-		await merchantWCP.disablePaymentMethod( UPE_METHOD_CHECKBOXES );
-		await merchant.logout();
-	} );
+const cardTestingPreventionStates = [
+	{ cardTestingPreventionEnabled: false },
+	{ cardTestingPreventionEnabled: true },
+];
 
-	describe.each( bnplProviders )( 'Checkout with %s', ( providerName ) => {
-		beforeEach( async () => {
-			await setupProductCheckout(
+describe.each( cardTestingPreventionStates )(
+	'BNPL checkout',
+	( { cardTestingPreventionEnabled } ) => {
+		beforeAll( async () => {
+			await merchant.login();
+			await merchantWCP.enablePaymentMethod( UPE_METHOD_CHECKBOXES );
+			if ( cardTestingPreventionEnabled ) {
+				await merchantWCP.enableCardTestingProtection();
+			}
+			await merchant.logout();
+			await shopper.login();
+			await shopperWCP.changeAccountCurrencyTo(
 				config.get( 'addresses.customer.billing' ),
-				[ [ 'Beanie', 3 ] ]
+				'USD'
 			);
 		} );
 
-		it( `should successfully place order with ${ providerName }`, async () => {
-			await uiUnblocked();
-			// Select BNPL provider as payment method.
-			const xPathPaymentMethodSelector = `//*[@id='payment']/ul/li/label[contains(text(), '${ providerName }')]`;
-			await page.waitForXPath( xPathPaymentMethodSelector );
-			const [ paymentMethodLabel ] = await page.$x(
-				xPathPaymentMethodSelector
-			);
-			await paymentMethodLabel.click();
-			await shopper.placeOrder();
+		afterAll( async () => {
+			await shopperWCP.logout();
+			await merchant.login();
+			await merchantWCP.disablePaymentMethod( UPE_METHOD_CHECKBOXES );
+			if ( cardTestingPreventionEnabled ) {
+				await merchantWCP.disableCardTestingProtection();
+			}
```

Comment:
Can we add `shopperWCP.emptyCart()` on `afterAll` since the test adds products to cart.

In case the test breaks, it might cause a cascade failure.

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/8164/commits/a07d017aa9c3852e1b23538fece0319c4c24b285.